azurerm_federated_credential in place updates fails with resource already exists error - import into state

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.14.8

### AzureRM Provider Version

4.75.0

### Affected Resource(s)/Data Source(s)

azurerm_federated_credentials

### Terraform Configuration Files

```hcl
resource "azurerm_federated_identity_credential" "xxx" {
  name                      = "xxx"
  user_assigned_identity_id = azurerm_user_assigned_identity.xxx.id
  issuer                    = var.aks_oidc_issuer_url
  subject                   = "system:serviceaccount:xxx"
  audience                  = ["api://AzureADTokenExchange"]
}
```

### Debug Output/Panic Output

```shell
2026-06-04T15:07:05.629+0100 [DEBUG] provider.terraform-provider-azurerm_v4.75.0_x5: {"id":"/subscriptions/<SUBSCRIPTION_ID>/resourcegroups/<RESOURCE_GROUP>/providers/Microsoft.ManagedIdentity/userAssignedIdentities/<UAI_NAME>/federatedIdentityCredentials/<FIC_NAME>","name":"<FIC_NAME>","type":"Microsoft.ManagedIdentity/userAssignedIdentities/federatedIdentityCredentials","properties":{"issuer":"https://uksouth.oic.prod-aks.azure.com/<TENANT_ID>/<CLUSTER_ID>/","subject":"system:serviceaccount:<K8S_NAMESPACE>:<SERVICE_ACCOUNT>","audiences":["api://AzureADTokenExchange"]}}
2026-06-04T15:07:05.629+0100 [DEBUG] provider.terraform-provider-azurerm_v4.75.0_x5: [DEBUG] Unlocking "/subscriptions/<SUBSCRIPTION_ID>/resourceGroups/<RESOURCE_GROUP>/providers/Microsoft.ManagedIdentity/userAssignedIdentities/<UAI_NAME>"
2026-06-04T15:07:05.629+0100 [DEBUG] provider.terraform-provider-azurerm_v4.75.0_x5: [DEBUG] Unlocked "/subscriptions/<SUBSCRIPTION_ID>/resourceGroups/<RESOURCE_GROUP>/providers/Microsoft.ManagedIdentity/userAssignedIdentities/<UAI_NAME>"
2026-06-04T15:07:05.629+0100 [TRACE] provider.terraform-provider-azurerm_v4.75.0_x5: Called downstream: @module=sdk.helper_schema tf_mux_provider="*schema.GRPCProviderServer" tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_req_id=<REQUEST_ID> tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.38.1/helper/schema/resource.go:991 tf_resource_type=azurerm_federated_identity_credential timestamp="2026-06-04T15:07:05.629+0100"
2026-06-04T15:07:05.629+0100 [TRACE] provider.terraform-provider-azurerm_v4.75.0_x5: Received downstream response: @caller=github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/internal/tf5serverlogging/downstream_request.go:61 @module=sdk.proto diagnostic_error_count=1 tf_proto_version=5.10 tf_req_duration_ms=1175 tf_rpc=ApplyResourceChange diagnostic_warning_count=0 tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_req_id=<REQUEST_ID> tf_resource_type=azurerm_federated_identity_credential timestamp="2026-06-04T15:07:05.629+0100"
2026-06-04T15:07:05.629+0100 [ERROR] provider.terraform-provider-azurerm_v4.75.0_x5: Response contains error diagnostic: tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/internal/diag/diagnostics.go:58 diagnostic_detail="a resource with the ID \"/subscriptions/<SUBSCRIPTION_ID>/resourceGroups/<RESOURCE_GROUP>/providers/Microsoft.ManagedIdentity/userAssignedIdentities/<UAI_NAME>/federatedIdentityCredentials/<FIC_NAME>\" already exists - to be managed via Terraform this resource needs to be imported into the State. Please see the resource documentation for \"azurerm_federated_identity_credential\" for more information" diagnostic_summary="a resource with the ID \"/subscriptions/<SUBSCRIPTION_ID>/resourceGroups/<RESOURCE_GROUP>/providers/Microsoft.ManagedIdentity/userAssignedIdentities/<UAI_NAME>/federatedIdentityCredentials/<FIC_NAME>\" already exists - to be managed via Terraform this resource needs to be imported into the State. Please see the resource documentation for \"azurerm_federated_identity_credential\" for more information" tf_proto_version=5.10 tf_provider_addr=registry.terraform.io/hashicorp/azurerm @module=sdk.proto diagnostic_severity=ERROR tf_req_id=<REQUEST_ID> tf_resource_type=azurerm_federated_identity_credential timestamp="2026-06-04T15:07:05.629+0100"
2026-06-04T15:07:05.629+0100 [TRACE] provider.terraform-provider-azurerm_v4.75.0_x5: Served request: tf_proto_version=5.10 tf_req_id=<REQUEST_ID> tf_resource_type=azurerm_federated_identity_credential @module=sdk.proto tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/tf5server/server.go:957 timestamp="2026-06-04T15:07:05.629+0100"
2026-06-04T15:07:05.629+0100 [DEBUG] provider.terraform-provider-azurerm_v4.75.0_x5: [DEBUG] Locked "/subscriptions/<SUBSCRIPTION_ID>/resourceGroups/<RESOURCE_GROUP>/providers/Microsoft.ManagedIdentity/userAssignedIdentities/<UAI_NAME>"
2026-06-04T15:07:05.629+0100 [TRACE] terraform.contextPlugins: Schema for provider "registry.terraform.io/hashicorp/azurerm" is in the global cache
```

### Expected Behaviour

```
  # module.infra.azurerm_federated_identity_credential.xxx["xxx"] will be updated in-place
  ~ resource "azurerm_federated_identity_credential" "xxx" {
        id                        = "/subscriptions/..."
      ~ issuer                    = "https://uksouth.oic.prod-aks.azure.com/old/" -> "https://uksouth.oic.prod-aks.azure.com/new/"
        name                      = "xxx"
        # (5 unchanged attributes hidden)
    }
```

The existing federated credential should've been updated in place as the plan suggests.

### Actual Behaviour

```
│ Error: a resource with the ID "/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.ManagedIdentity/userAssignedIdentities/xxx/federatedIdentityCredentials/xxx" already exists - to be managed via Terraform this resource needs to be imported into the State. Please see the resource documentation for "azurerm_federated_identity_credential" for more information
│ 
│   with module.infra.azurerm_federated_identity_credential.xxx["xxx"],
│   on .terraform/modules/infra/uai_sb_receiver.tf line 8, in resource "azurerm_federated_identity_credential" "xxx":
│    8: resource "azurerm_federated_identity_credential" "xxx" {
│ 
│ a resource with the ID "/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.ManagedIdentity/userAssignedIdentities/xxx/federatedIdentityCredentials/xxx" already exists - to be managed via Terraform this resource needs to be imported into the State. Please see the resource documentation for "azurerm_federated_identity_credential" for more information
```

### Steps to Reproduce

Deploy a fed cred, change the issuer_url on it (perhaps other vars also) and see it break.

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/managedidentity/federated_identity_credential_resource.go
+++ b/internal/services/managedidentity/federated_identity_credential_resource.go
@@ -49,33 +49,34 @@ func (r FederatedIdentityCredentialResource) ResourceType() string {
 
 func (r FederatedIdentityCredentialResource) Arguments() map[string]*pluginsdk.Schema {
 	schema := map[string]*pluginsdk.Schema{
+		"name": {
+			Required: true,
+			ForceNew: true,
+			Type:     pluginsdk.TypeString,
+		},
+
+		"user_assigned_identity_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateUserAssignedIdentityID,
+		},
+
 		"audience": {
 			Elem: &pluginsdk.Schema{
 				Type: pluginsdk.TypeString,
 			},
-			ForceNew: false,
 			Required: true,
 			Type:     pluginsdk.TypeList,
 			MaxItems: 1,
 		},
+
 		"issuer": {
-			ForceNew: false,
 			Required: true,
 			Type:     pluginsdk.TypeString,
 		},
-		"name": {
-			ForceNew: true,
-			Required: true,
-			Type:     pluginsdk.TypeString,
-		},
-		"user_assigned_identity_id": {
-			Type:         pluginsdk.TypeString,
-			ForceNew:     true,
-			Required:     true,
-			ValidateFunc: commonids.ValidateUserAssignedIdentityID,
-		},
+
 		"subject": {
-			ForceNew: false,
 			Required: true,
 			Type:     pluginsdk.TypeString,
 		},
@@ -202,7 +203,58 @@ func (r FederatedIdentityCredentialResource) Read() sdk.ResourceFunc {
 }
 
 func (r FederatedIdentityCredentialResource) Update() sdk.ResourceFunc {
-	return r.Create()
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, rmd sdk.ResourceMetaData) error {
+			client := rmd.Client.ManagedIdentity.V20241130.FederatedIdentityCredentials
+
+			var config FederatedIdentityCredentialResourceSchema
+			if err := rmd.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			id, err := federatedidentitycredentials.ParseFederatedIdentityCredentialID(rmd.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			if existing.Model == nil {
+				return fmt.Errorf("retrieving %s: `model` was nil`", id)
+			}
+
+			if existing.Model.Properties == nil {
+				return fmt.Errorf("retrieving %s: `properties` was nil", id)
+			}
+			props := existing.Model.Properties
+
+			if rmd.ResourceData.HasChange("audience") {
+				props.Audiences = config.Audience
+			}
+
+			if rmd.ResourceData.HasChange("issuer") {
+				props.Issuer = config.Issuer
+			}
+
+			if rmd.ResourceData.HasChange("subject") {
+				props.Subject = config.Subject
+			}
+
+			parentId := commonids.NewUserAssignedIdentityID(id.SubscriptionId, id.ResourceGroupName, id.UserAssignedIdentityName)
+			locks.ByID(parentId.ID())
+			defer locks.UnlockByID(parentId.ID())
+
+			if _, err := client.CreateOrUpdate(ctx, *id, *existing.Model); err != nil {
+				return fmt.Errorf("updating %s: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
 }
 
 func (r FederatedIdentityCredentialResource) Delete() sdk.ResourceFunc {

--- a/internal/services/managedidentity/federated_identity_credential_resource_test.go
+++ b/internal/services/managedidentity/federated_identity_credential_resource_test.go
@@ -47,6 +47,14 @@ func TestAccFederatedIdentityCredential_basic(t *testing.T) {
 				check.That(data.ResourceName).Key("subject").MatchesRegex(&rg),
 			),
 		},
+		data.ImportStep(),
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
 	})
 }
 

